execution/executor-benchmark/src/lib.rs

```diff
@@ -141,18 +161,23 @@ impl TransactionGenerator {
                 );
                 transactions.push(txn);
             }
-
-            self.block_sender
-                .as_ref()
-                .unwrap()
-                .send(transactions)
-                .unwrap();
+            if let Some(sender) = &self.block_sender {
+                sender.send(transactions).unwrap();
+            } else {
+                txn_block.push(transactions);
+            }
```

Comment:
Draining the channel into a Vec on in the benchmark code works too (better to me). But I'm fine with this as well.

Reply:
I'm a bit concerned about the overhead of having to spawn a new thread in the bencher.